gobblin-service/src/main/java/org/apache/gobblin/service/modules/orchestration/proc/DagProcUtils.java

```diff
@@ -199,6 +202,21 @@ private static void sendCancellationEvent(Dag.DagNode<JobExecutionPlan> dagNodeT
     jobExecutionPlan.setExecutionStatus(CANCELLED);
   }
 
+  public static void setAndEmitFlowEvent(EventSubmitter eventSubmitter, Dag<JobExecutionPlan> dag, String flowEvent) {
```

Comment:
this is little different to remove confusing code of `DagManagerUtils::emitFlowEvent` which will be cleaned up once DagProcEngine is stable